cli/disk_cache.rs

```diff
@@ -37,13 +37,48 @@ impl DiskCache {
           None => host.to_string(),
         };
         out.push(host_port);
+
+        for path_seg in url.path_segments().unwrap() {
+          out.push(path_seg);
+        }
+      }
+      "file" => {
+        let path = url.to_file_path().unwrap();
+        let mut path_components = path.components();
+
+        if cfg!(target_os = "windows") {
+          if let Some(Component::Prefix(prefix_component)) =
+            path_components.next()
+          {
+            // Windows doesn't support ":" in filenames, so we need to extract disk prefix
+            // Example: file:///C:/deno/js/unit_test_runner.ts
+            // it should produce: file\c\deno\js\unit_test_runner.ts
+            match prefix_component.kind() {
+              Prefix::Disk(disk_byte) | Prefix::VerbatimDisk(disk_byte) => {
+                let disk = (disk_byte as char).to_string();
+                out.push(disk);
+              }
+              _ => {}
```

Comment:
Add an unreachable() here. Let's not silently discard part components.

Reply:
@piscisaureus added